### PR TITLE
Make Matching Solutions Editable

### PR DESCRIPTION
**Pull request by florian-papsdorf**
Until now it was only possible to create or delete a matching solution. Now we can update information related to a matching solution.

### CODE REVIEW

File: app/src/store/actions/AlgorithmDialogStoreActions.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+} from 'utils/statusMessages';
+
+export const openAddDialog = (): easyPrimitiveActionReturn =>
+  easyPrimitiveAction({
+    type: DialogActions.OPEN_ADD_DIALOG,
+    // reducer ignores payload
+    payload: false,
+  });
+
+export const openChangeDialog = (
```
Comment (phpfs) [inline app/src/store/actions/AlgorithmDialogStoreActions.ts:24]: Equally add a openAddDialog() action which will reset the dialog?

**At the moment, after you edited a algorithm and click on the add button again, the fields are not reset.**

Maybe generally delete all contents if an edit is canceled? We only want to keep the input for the add case I guess :)

File: app/src/store/reducers/AlgorithmDialogReducer.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      };
+    case DialogActions.OPEN_CHANGE_DIALOG:
+      return {
+        ...state,
+        isOpen: true,
+        algorithmId: (action.payload as Algorithm).id,
+        algorithmName: (action.payload as Algorithm).name,
+        algorithmDescription: (action.payload as Algorithm).description ?? '',
+      };
+    case DialogActions.CLOSE_DIALOG:
```
Comment (phpfs) [inline app/src/store/reducers/AlgorithmDialogReducer.ts:31]: Maybe add reset here if closing an changeAlgorithmDialog?

File: app/src/utils/statusMessages.ts [hunk 3]
```diff
@@ -16,3 +16,4 @@ export const SUCCESS_LOAD_METRICS_TUPLES =
   'Successfully loaded metrics tuples!';
 export const SUCCESS_LOAD_BINARY_METRICS =
   'Successfully loaded binary metrics!';
+export const SUCCESS_UPDATE_ALGORITHM = 'Successfully updated algorithm';
```
Comment (phpfs) [inline app/src/utils/statusMessages.ts:19]: We generally do an "!" after our messages :)

**Review by phpfs — CHANGES_REQUESTED** [review]

**Comment by phpfs** [comment]
Really nice implementation - and easy to use as a template for further implementations! :)

**Comment by phpfs** [comment]
Btw - I updates the visuals - hope this is fine with you :)

File: app/src/utils/statusMessages.ts (hunk 3 above)
Comment (phpfs) [reply app/src/utils/statusMessages.ts:19]: Maybe 'Successfully updated the designated matching solution!'?

File: app/src/store/actions/AlgorithmDialogStoreActions.ts (hunk 1 above)
Comment (florian-papsdorf) [reply app/src/store/actions/AlgorithmDialogStoreActions.ts:24]: Of course, you are totally right. It should be a concern of the reducer.
Fix in commit 3755b26d

File: app/src/store/reducers/AlgorithmDialogReducer.ts (hunk 2 above)
Comment (florian-papsdorf) [reply app/src/store/reducers/AlgorithmDialogReducer.ts:31]: Yes, I added it here as a concern of the reducer.

File: app/src/utils/statusMessages.ts (hunk 3 above)
Comment (florian-papsdorf) [reply app/src/utils/statusMessages.ts]: Fine!

**Comment by phpfs** [comment]
This PR is part of #18.

**Pull request merged**